molgenis-data-import/src/main/java/org/molgenis/data/importer/emx/EmxMetaDataParser.java

```diff
@@ -853,70 +867,75 @@ private void parseAttributesSheet(Repository<Entity> attributesRepo, Intermediat
 				if (!emxIdAttrValue.equalsIgnoreCase("true") && !emxIdAttrValue.equalsIgnoreCase("false")
 						&& !emxIdAttrValue.equalsIgnoreCase(AUTO))
 				{
-					throw new IllegalArgumentException(
-							format("Attributes error on line [%d]. Illegal idAttribute value. Allowed values are 'TRUE', 'FALSE' or 'AUTO'",
-									rowIndex));
+					throw new InvalidAttributeValueException("idAttribute", emxIdAttrValue, "attributes",
+							new String[] { "TRUE", "FALSE", "AUTO" }, rowIndex);
 				}
 				if (emxIdAttrValue.equalsIgnoreCase("true"))
 				{
 					if (!isIdAttributeTypeAllowed(attr))
 					{
-						throw new MolgenisDataException("Identifier is of type [" + attr.getDataType()
-								+ "]. Id attributes can only be of type 'STRING', 'INT' or 'LONG'");
+						throw new InvalidIdentifierAttributeTypeException(attr.getDataType());
 					}
 				}
 
 				attr.setAuto(emxIdAttrValue.equalsIgnoreCase(AUTO));
-				if (!attr.isAuto())
-					emxAttr.setIdAttr(parseBoolean(emxIdAttrValue, rowIndex, EMX_ATTRIBUTES_ID_ATTRIBUTE));
+				if (!attr.isAuto()) emxAttr.setIdAttr(
+						parseBoolean(emxIdAttrValue, rowIndex, EMX_ATTRIBUTES_ID_ATTRIBUTE, "attributes"));
 				else emxAttr.setIdAttr(true); // If it is auto, set idAttr to true
 			}
 
 			if (attr.isAuto() && !isStringType(attr))
 			{
-				throw new IllegalArgumentException(
-						format("Attributes error on line [%d]. Auto attributes can only be of data type 'string'",
-								rowIndex));
+				throw new InvalidAttributeTypeException("auto", attr.getDataType(), attr.getName(),
+						new String[] { "string" }, rowIndex);
 			}
 			if (emxAttrVisible != null)
 			{
 				if (emxAttrVisible.equalsIgnoreCase("true") || emxAttrVisible.equalsIgnoreCase("false"))
 				{
-					attr.setVisible(parseBoolean(emxAttrVisible, rowIndex, EMX_ATTRIBUTES_VISIBLE));
+					attr.setVisible(parseBoolean(emxAttrVisible, rowIndex, EMX_ATTRIBUTES_VISIBLE, "attributes"));
 				}
 				else
 				{
 					attr.setVisibleExpression(emxAttrVisible);
 				}
 			}
-			if (emxAggregatable != null)
-				attr.setAggregatable(parseBoolean(emxAggregatable, rowIndex, EMX_ATTRIBUTES_AGGREGATEABLE));
-			if (emxReadOnly != null) attr.setReadOnly(parseBoolean(emxReadOnly, rowIndex, EMX_ATTRIBUTES_READ_ONLY));
-			if (emxUnique != null) attr.setUnique(parseBoolean(emxUnique, rowIndex, EMX_ATTRIBUTES_UNIQUE));
+			if (emxAggregatable != null) attr.setAggregatable(
```

Comment:
How come layout has changed?

Reply:
No clue, short investigation suggests the new layout is the correct one